test/test-connect-callback.js

```diff
@@ -0,0 +1,76 @@
+var assert = require('assert');
+var fs = require('fs');
+
+var unix = require('../lib/unix_dgram');
+var SOCKNAME = '/tmp/unix_dgram.sock';
+
+var sentCount = 0;
+var seenCount = 0;
+var expected = 300;
+
+process.on('exit', function() {
+  assert.equal(seenCount, sentCount);
+});
+
+try { fs.unlinkSync(SOCKNAME); } catch (e) { /* swallow */ }
+
+var server = unix.createSocket('unix_dgram', function(buf, rinfo) {
+  console.error('server recv', '' + buf, arguments);
+  assert.equal('' + buf, 'PING' + seenCount);
+  if (++ seenCount === expected) {
+    server.close();
+    client.close();
+  }
+});
+server.bind(SOCKNAME);
+
+var client = unix.createSocket('unix_dgram', function(buf, rinfo) {
+  console.error('client recv', arguments);
+  assert(0);
+});
+
+client.on('error', function(err) {
+  console.error(err);
+  assert(0);
+});
+
+client.on('connect', function() {
+  console.error('connected');
+
+  client.on('congestion', function() {
+    console.error('congestion');
+    throw new Error('Should not emit congestion');
+  });
+
+  client.on('writable', function() {
+    console.error('writable');
+  });
+
+  /*
+   * Usually /proc/sys/net/unix/max_dgram_qlen is 10 so one that's being processed, 10 in the
+   * recv queue and the ... is dropped
+   */
```

Comment:
To be honest, I copied this comment from the test-connect.js file, without having being able to see it in action